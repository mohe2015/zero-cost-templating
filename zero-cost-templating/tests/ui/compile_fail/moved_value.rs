#![feature(coroutines)]

extern crate alloc;

use alloc::borrow::Cow;

use zero_cost_templating_macros::template_stream;

#[template_stream("test.html.hbs")]
pub async fn test() {
    let template = test_initial0!();
    let _ = test_template0!(template);
    let template = test_template0!(template);
    let page_title = Cow::from("thetitle");
    let template = test_page_title1!(template, page_title);
    let csrf_token = Cow::from("thetoken");
    let template = test_csrf_token2!(template, csrf_token);
    let template = test_template6!(template);
    test_copyright_year7!(template, Cow::from("2023".to_string()));
}

pub fn main() {}
