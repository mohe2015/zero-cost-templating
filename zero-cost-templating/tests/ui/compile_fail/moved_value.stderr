error[E0382]: use of moved value: `template`
  --> tests/ui/compile_fail/moved_value.rs:13:31
   |
11 |     let template = initial0!();
   |         -------- move occurs because `template` has type `TestTemplate0`, which does not implement the `Copy` trait
12 |     let _ = template0!(template);
   |                        -------- value moved here
13 |     let template = template0!(template);
   |                               ^^^^^^^^ value used here after move
